library/std/src/io/mod.rs

```diff
@@ -474,11 +470,6 @@ pub(crate) fn default_read_to_end<R: Read + ?Sized>(
         spare = &mut spare[..buf_len];
         let mut read_buf: BorrowedBuf<'_> = spare.into();
 
-        // SAFETY: These bytes were initialized but not filled in the previous loop
-        unsafe {
-            read_buf.set_init(initialized);
-        }
-
         let mut cursor = read_buf.unfilled();
```

Comment:
Same here, this will be a perf regression for slow readers (eg if it returns one byte at a time).